pennylane/ops/channel.py

```diff
@@ -563,7 +565,8 @@ class PauliError(Channel):
     num_params = 2
     """int: Number of trainable parameters that the operator depends on."""
 
-    def __init__(self, operators, p, wires=None, id=None):
+    def __init__(self, operators, p, wires: WiresLike, id=None):
```

Comment:
So here (and in several places below) we are no longer providing a default value for `wires`. Therefore, I assume that if someone was using this part of the code without providing the wires, it would no longer work after this change. Isn't this a breaking change? Please let me know if I am missing something

Reply:
On master,
```python
>>> qml.PauliError("X", 0.5)
"ValueError: Must specify the wires that PauliError acts on"
```
with a long traceback. 

With this PR we get,
```python
>>> qml.PauliError("X", 0.5)
TypeError: PauliError.__init__() missing 1 required positional argument: 'wires'
```

Although I'm not sure why one wouldn't specify the wires for this channel? Would you consider this a breaking change if the error message is different?

Reply:
Likewise, for `QubitChannel`, prior to this PR, we get this ambiguous error,
```python
import pennylane as qml
import numpy as np

K_list1 = [
    np.array([[1.0, 0.0], [0.0, 0.9486833]]),
    np.array([[0.0, 0.31622777], [0.0, 0.0]]),
]

>>> out = qml.QubitChannel(K_list1).kraus_matrices()
WireError: Wires must be hashable; got [[0.         0.31622777]
 [0.         0.        ]].
```

but with this PR we get,
```
TypeError: QubitChannel.__init__() missing 1 required positional argument: 'wires'
```

Thoughts?

Reply:
So basically the default value that was provided for `wires` was not working :sweat_smile: 

Thanks again for fixing this :1st_place_medal: 

The only observation left would be to add a changelog entry since we are modifying the source code

Reply:
Okay, will do!

Reply:
Changelog has been updated 😎